hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/signschedule/SignScheduleTranslator.java

```diff
@@ -16,32 +16,171 @@
+     * @return the transaction body
+     */
+    private TransactionBody bodyFor(@NonNull ScheduleID scheduleID) {
+        requireNonNull(scheduleID);
+        return TransactionBody.newBuilder()
+                .scheduleSign(ScheduleSignTransactionBody.newBuilder()
+                        .scheduleID(scheduleID)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Extracts the schedule ID from a {@code authorizeSchedule()} call or return the redirect schedule ID
+     * if the call via the proxy contract
+     *
+     * @param attempt the call attempt
+     * @return the schedule ID
+     */
+    private ScheduleID scheduleIdFor(@NonNull HssCallAttempt attempt) {
+        requireNonNull(attempt);
+        if (attempt.isSelector(SIGN_SCHEDULE_PROXY)) {
+            return getScheduleIDForSignSchedule(attempt);
+        } else if (attempt.isSelector(AUTHORIZE_SCHEDULE)) {
+            return getScheduleIDForAuthorizeSchedule(attempt);
+        }
+        throw new IllegalStateException("Unexpected function selector");
+    }
+
+    private static ScheduleID getScheduleIDForSignSchedule(@NonNull HssCallAttempt attempt) {
+        final var scheduleID = attempt.redirectScheduleId();
+        validateTrue(scheduleID != null, INVALID_SCHEDULE_ID);
+        return attempt.redirectScheduleId();
+    }
+
+    private static ScheduleID getScheduleIDForAuthorizeSchedule(@NonNull HssCallAttempt attempt) {
+        final var call = AUTHORIZE_SCHEDULE.decodeCall(attempt.inputBytes());
+        final Address scheduleAddress = call.get(0);
+        final var number = numberOfLongZero(explicitFromHeadlong(scheduleAddress));
+        final var schedule = attempt.enhancement().nativeOperations().getSchedule(number);
+        validateTrue(schedule != null, INVALID_SCHEDULE_ID);
+        return schedule.scheduleId();
+    }
+
+    /**
+     * Extracts the key set for a {@code signSchedule()} call.
+     *
+     * @param attempt the call attempt
+     * @return the key set
+     */
+    private Set<Key> keySetFor(@NonNull HssCallAttempt attempt) {
+        requireNonNull(attempt);
+        if (attempt.isSelector(SIGN_SCHEDULE_PROXY)) {
+            return getKeysForSignSchedule(attempt);
+        } else if (attempt.isSelector(AUTHORIZE_SCHEDULE)) {
+            return getKeysForAuthorizeSchedule(attempt);
+        }
+        throw new IllegalStateException("Unexpected function selector");
+    }
+
+    @NonNull
+    private static Set<Key> getKeysForSignSchedule(@NonNull HssCallAttempt attempt) {
+        // If an Eth sender key is present, use it. Otherwise, use the account key if present.
+        Key key = attempt.enhancement().systemOperations().maybeEthSenderKey();
+        if (key != null) {
+            return Set.of(key);
+        }
+        key = attempt.enhancement().nativeOperations().getAccountKey(attempt.originatorAccount());
+        if (key != null) {
+            return Set.of(key);
```

Comment:
What if this account key is not a simple key.  Isn't this eventually what is sent to `ChildDispatchFactory` where it becomes what's returned by a key verifiers `authorizingSimpleKeys`?  Wouldn't that mean it needs to be a simple key?

Reply:
I image that the key will be passed into the `verifier` and as long as the `verifier` can handle key lists for verification then it should work?  It looks like the `DefaultKeyVerifier` which is what I am passing does allow verification of things like key lists but not sure here.  @tinker-michaelj ?